frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -45,9 +43,9 @@ const CloudShellTerminal: React.FC<CloudShellTerminalProps> = ({ username, onCan
   }
 
   if (Array.isArray(data)) {
-    const workspace = data.find(
-      (d) => d?.metadata?.annotations?.[CLOUD_SHELL_USER_ANNOTATION] === username,
-    );
+    const workspace = isKubeAdmin
+      ? data.find((d) => d?.metadata?.labels?.[CLOUD_SHELL_CREATOR_LABEL] === '')
```

Comment:
You should be using the `matchLabels` selector in the `useK8sWatchResource` hook. Search for the label match `CLOUD_SHELL_CREATOR_LABEL`.
This way you no longer need to loop over the results and instead you can use the first result.

Reply:
Guys, seems we forgot to mention really important thing.
You still should filter all workspaces which are mutable. So you should check `org.eclipse.che.workspace/immutable: "true"` annotation.

Reply:
Just a bit more info about immutable annotation: It guarantees that nobody who has edit rights for workspace CR will patch your terminal workspace and put custom dev container or cheEditor that will steal user's token

Reply:
@vikram-raj keep the loop we had already for the username check and now add this immutable annotation check.